src/ripple/basics/Slice.h

```diff
@@ -117,6 +126,21 @@ class Slice
     }
     /** @} */
 
+    /** Shrinks the slice by moving its start forward by n characters. */
+    void
+    remove_prefix(std::size_t n)
```

Comment:
I opted to duplicate the interface of `std::string_view` since `Slice` is, by any other name, a `std::string_view`. The intention is to eventually eliminate the need for `Slice` altogether.

Reply:
Can be noexcept

Reply:
Can be 
`    constexpr void
    remove_prefix(std::size_t n) noexcept`

Reply:
Why not eliminate this class and use `std::string_view`?

Reply:
I'm not sure I agree that `Slice` should be replaced `string_view`; `Slice` is for bytes (opaque binary data), `string_view` is for strings. We do sometimes use string-like containers for binary data, but I always think that sends the wrong signal as-to what type of data is stored. We probably want `std::span`, but that doesn't come until c++-20.

Having said that, it's only a weak objection. If both of you want to use `string_view` for this I'm OK with it.

Reply:
@seelabs good point and convincing enough for me to stay as is until `span`.

Reply:
I agree that `span` is closer to our semantics.  But I don't yet have the experience to know that `span`'s API is a good fit for replacing `Splice`.  It _probably_ is.  If it is, we should be leaning towards that API as it is different than `string_view`.

Reply:
There's nothing about `std::string_view`, as far as I can tell, that requires it be used for strings, except perhaps its unfortunate name. I believe it was misnamed and should have been `std::byte_sequence` or somesuch, with a std aliases for a `const char*` instantiation and/or a specialization for `std::string`. 

We could replace `Slice` with `string_view` now but I don't know if that's a good idea. I agree that using `string_view` as the name implies something that may not be true for arbitrary binary blobs. Beyond the bad name it has a dangerous API, which I'm surprised survived review: some of the APIs respect boundaries, others don't. 

My hope is that we can remove custom code in favor of more `std::` stuff; the less code we need to write and maintain the better, especially for boilerplate stuff like this, but I don't know that we have a good `std::` alternative.

We could use [`boost::asio::const_buffer`](https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/const_buffer.html) and [`boost::asio::mutable_buffer`](https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/mutable_buffer.html) instead, but they lack some the nice functions that `std::string_view` offers.

Maybe we can just punt on further changes to `Slice` for now until we find something good.